### PR TITLE
Preapps 1126 2 Show GAL as a "System Folder" in contacts

### PR DESCRIPTION
**Pull request by damini-vashishtha**

### CODE REVIEW

File: src/schema/schema.graphql [hunk 1]
```diff
@@ -2175,8 +2189,12 @@ type Query {
 		channel: SetRecoveryAccountChannel!
 	): RecoverAccount
 	relatedContacts(email: String!): [RelatedContact]
-	shareInfos(addresses: [String!]): [[ShareInfo]]
-
+	shareInfos(
```
Comment (silentsakky) [inline src/schema/schema.graphql]: shareInfo

File: src/batch-client/types.ts [hunk 2]
```diff
@@ -138,7 +140,10 @@ export interface SearchOptions {
 }
 
 export interface ShareInfosOptions {
```
Comment (silentsakky) [inline src/batch-client/types.ts]: shareInfoOptions

File: src/batch-client/index.ts [hunk 3]
```diff
@@ -846,22 +846,14 @@ export class ZimbraBatchClient {
...
-						owner: {
-							by: 'name',
-							_content: address
-						},
-						_jsns: 'urn:zimbraAccount'
-					}
-				})
-			)
-		);
+	public shareInfos = (options: ShareInfosOptions) =>
```
Comment (silentsakky) [inline src/batch-client/index.ts]: public shareInfo = (options: ShareInfoOptions) =>

File: src/schema/schema.graphql [hunk 4]
```diff
@@ -2060,6 +2062,18 @@ input DismissInput {
 	dismissedAt: Float!
 }
 
+
+input Grantee {
+	granteeId: String
```
Comment (silentsakky) [inline src/schema/schema.graphql]: input parameters are as below
type, id, name

Comment (silentsakky) [inline src/schema/schema.graphql]: id: ID

File: src/batch-client/types.ts (hunk 2 above)
Comment (damini-vashishtha) [reply src/batch-client/types.ts]: should I change the first letter to small 's'?

Comment (silentsakky) [reply src/batch-client/types.ts]: my bad it should be ShareInfoOptions
change plural to singular form

File: src/schema/schema.graphql (hunk 1 above)
Comment (damini-vashishtha) [reply src/schema/schema.graphql]: Done

File: src/batch-client/types.ts (hunk 2 above)
Comment (damini-vashishtha) [reply src/batch-client/types.ts]: Done

File: src/batch-client/index.ts (hunk 3 above)
Comment (damini-vashishtha) [reply src/batch-client/index.ts]: Done

File: src/schema/schema.graphql (hunk 4 above)
Comment (damini-vashishtha) [reply src/schema/schema.graphql]: Done

File: src/schema/schema.graphql [hunk 5]
```diff
@@ -1306,6 +1307,7 @@ type ShareInfo {
 	ownerName: String
 	rights: String
 	view: FolderView
+	mid: String
```
Comment (itsmeprashant) [inline src/schema/schema.graphql]: type ID

Comment (damini-vashishtha) [reply src/schema/schema.graphql]: Done

**Pull request merged**